Added an overload of RespondAsync to IRpc that allows the responder to read the message headers

c0c0bird opened the pull request:
RequestAsync already allows the producer to add headers.
On the consumer/responder side, those headers need to be processed.
This overload allows the responder action to accept MessageProperties.

Pliner commented:
Hey @c0c0bird,

Thanks for your contribution. There are some comments from me:
1. I reckon it is better not to create one more method in IRpc interface, but replace existing Respond method by it + add necessary extensions to cover signature of the existing Respond method.
2. I am in doubts about an addition of MessageProperties. Why not to add MessageReceivedInfo and Body? Or add only headers?
3. We need to understand how to do the same for ISendReceive and IPubSub.

c0c0bird commented:
1. Ok, you are right. I moved the old signature to the extensions.
2. Ok, I changed the method to use only the headers dictionary. I do not really like that, because given only the signature, it is unclear to the user what the new parameter does. A generic dictionary could be anything. Just the comment explains, that it will contain the headers.
3. ISendReceive and IPubSub should work quite similar. Anyway, I think they are not that importat, because users can quite easily switch to AdvancedBus.
